contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -2516,6 +2516,36 @@ $$
 LANGUAGE 'pltsql';
 GRANT EXECUTE ON PROCEDURE sys.sp_helpsrvrolemember TO PUBLIC;
 
+CREATE OR REPLACE PROCEDURE sys.sp_helpdbfixedrole("@rolename" sys.SYSNAME = NULL) AS
+$$
+DECLARE
+	@trimmedrole sys.SYSNAME = LOWER(RTRIM(@rolename));
+BEGIN
+	-- Returns a list of the fixed database roles. 
+	-- Only fixed role present in babelfish is db_owner.
+	IF @trimmedrole IS NULL OR @trimmedrole = 'db_owner'
+	BEGIN
+		SELECT * 
+		FROM (VALUES (CAST('db_owner' AS sys.SYSNAME), CAST('DB Owners' AS sys.nvarchar(70))))
+			AS t (DbFixedRole, Description)
+	END
+	ELSE IF @trimmedrole IN (
+			'db_accessadmin','db_securityadmin','db_ddladmin', 'db_backupoperator', 
+			'db_datareader', 'db_datawriter', 'db_denydatareader', 'db_denydatawriter')
+	BEGIN
+		-- Return an empty result set instead of raising an error
+		SELECT * 
+		FROM (VALUES (CAST(NULL AS sys.SYSNAME), CAST(NULL AS sys.nvarchar(70))))
+			AS t (DbFixedRole, Description) 
+		WHERE 1=0	
+	END
```

Comment:
Is there a better way to return an empty result set?

Reply:
That's a good question and I don't know the answer... for now I think this is acceptable